File: exercises/D701/solution.java
import java.util.*;

class D701 {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int q[] = new int[1000];
        int qs = 0, qe = 0;
        int n = in.nextInt();
        for (int i = 0; i < n; i++) {
            String s = in.next();
            if (s.equals("PUSH")) {
                int x = in.nextInt();
                q[qe++] = x;
            } else if (s.equals("FRONT")) {
                if (qs == qe) {
                    System.out.println("Empty");
                } else {
                    System.out.println(q[qs]);
                }
            } else if (s.equals("POP")) {
                if (qs == qe) {
                    System.out.println("Cannot pop");
                } else {
                    qs++;
                }
            } else {
                System.out.println(qe - qs);
            }
        }
    }
}

File: exercises/D701/solution.py
n = int(input())
qs = 0
qe = 0
q = []
for _ in range(n):
    args = input().split()
    if args[0] == 'PUSH':
        q.append(args[1])
        qe += 1
    elif args[0] == 'FRONT':
        if qs == qe:
            print('Empty')
        else:
            print(q[qs])
    elif args[0] == 'POP':
        if qs == qe:
            print('Cannot pop')
        else:
            qs += 1
    else:
        print(qe - qs)
